src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -18,273 +18,375 @@
+    static_assert(to_underlying(NetworkCommissioningStatus::kRegulatoryError       ) == to_underlying(Status::kRegulatoryError       ), "kRegulatoryError value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kAuthFailure           ) == to_underlying(Status::kAuthFailure           ), "kAuthFailure value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kUnsupportedSecurity   ) == to_underlying(Status::kUnsupportedSecurity   ), "kUnsupportedSecurity value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kOtherConnectionFailure) == to_underlying(Status::kOtherConnectionFailure), "kOtherConnectionFailure value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kIPV6Failed            ) == to_underlying(Status::kIPV6Failed            ), "kIPV6Failed value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kIPBindFailed          ) == to_underlying(Status::kIPBindFailed          ), "kIPBindFailed value mismatch.");
+    static_assert(to_underlying(NetworkCommissioningStatus::kUnknownError          ) == to_underlying(Status::kUnknownError          ), "kUnknownError value mismatch.");
+    // clang-format on
+    return static_cast<NetworkCommissioningStatus>(to_underlying(status));
+}
+
+} // namespace
 
-struct WiFiNetworkInfo
+CHIP_ERROR Instance::Register()
 {
-    uint8_t mSSID[kMaxWiFiSSIDLen + 1];
-    uint8_t mSSIDLen;
-    uint8_t mCredentials[kMaxWiFiCredentialsLen];
-    uint8_t mCredentialsLen;
-};
+    ReturnErrorOnFailure(chip::app::InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+    ReturnErrorOnFailure(
+        DeviceLayer::PlatformMgrImpl().AddEventHandler(_OnCommissioningComplete, reinterpret_cast<intptr_t>(this)));
+    return CHIP_NO_ERROR;
+}
 
-struct NetworkInfo
+void Instance::InvokeCommand(HandlerContext & ctxt)
 {
-    uint8_t mNetworkID[kMaxNetworkIDLen];
-    uint8_t mNetworkIDLen;
-    uint8_t mEnabled;
-    NetworkType mNetworkType;
-    union NetworkData
+    if (mAsyncCommandHandle.Get() != nullptr)
     {
-#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
-        Thread::OperationalDataset mThread;
-#endif
-#if defined(CHIP_DEVICE_LAYER_TARGET)
-        WiFiNetworkInfo mWiFi;
-#endif
-    } mData;
-};
+        // We have a command processing in the backend, reject all incoming commands.
+        ctxt.mCommandHandler.AddStatus(ctxt.mRequestPath, Protocols::InteractionModel::Status::Busy);
+        ctxt.SetCommandHandled();
+        return;
+    }
 
-namespace {
-// The internal network info containing credentials. Need to find some better place to save these info.
-NetworkInfo sNetworks[kMaxNetworks];
-} // namespace
+    // We only use mAsyncCommandHandle to determine whether we are processing another command, so it is OK to left some values
+    // dirty.
```

Comment:
Can this comment be improved to explain exactly how this works?  What values exactly are left dirty, and what does that mean?

Reply:
Comment updated.